torch_xla/experimental/pjrt.py

```diff
@@ -218,6 +226,45 @@ def _discover_tpu_master_worker_ip(device: torch.device):
   return _merge_replica_results(replica_results)
 
 
+@requires_pjrt
+def _run_singleprocess(fn: Callable[..., R],
+                       *args,
+                       start_method: str = 'spawn',
+                       master_port: int = 12355,
+                       **kwargs) -> Dict[int, R]:
+  """Runs `fn` on a single device core.
+
+  Spawns one process on a single physical device (e.g. TPU chip).
+
+  Args:
+    fn: Function to run on the device devices
+    args: args to pass to `fn`
+    start_method: The Python `multiprocessing` process creation method.
+      Default: `spawn`
+    kwargs: kwargs to pass to `fn`
+
+  Returns:
+    the result of calling `fn`.
+  """
+  os.environ.setdefault('LOCAL_WORLD_SIZE', '1')
+
+  if device_type() == 'TPU':
+    tpu.configure_one_chip_topology()
+
+  device = xm.get_xla_supported_devices()[0]
+  xm.set_replication(xm.xla_device(), [])
+
+  @functools.wraps(fn)
+  def _thread_fn(device: torch.device):
```

Comment:
You can remove `thread_fn` and call `fn()` directly. `set_replication` already sets the default device correctly in this case: https://github.com/pytorch/xla/blob/4c7c343c8482df80c21a365e2a945713ff0cf04e/torch_xla/core/xla_model.py#L345